src/software/multithreading/threaded_observer.h

```diff
@@ -12,13 +12,13 @@
  *
  * @tparam T The type of object this class is observing
  */
-template <typename T>
-class ThreadedObserver : public Observer<T>
+template <typename T, bool Ordered>
```

Comment:
Can we use an `enum` to indicate if it's ordered or not? Will be way more readable than a random bool.

Reply:
done, template non-type paramter is an `enum class` now